dev/AccessControl/GetSecurityDescriptorForAppContainerNames.cpp

```diff
@@ -0,0 +1,100 @@
+﻿#include <pch.h>
+#include <wil/resource.h>
+#include <wil/result.h>
+#include <accctrl.h>
+#include <aclapi.h>
+#include <userenv.h>
+#include "Security.AccessControl.h"
+#include "SecurityDescriptorHelpers.h"
+#include <vector>
+
+using namespace std;
+
+STDAPI GetSecurityDescriptorForAppContainerNames(
+    uint32_t accessRequestCount,
+    _In_reads_(accessRequestCount)
+    const AppContainerNameAndAccess* accessRequests,
+    _In_opt_ PSID principal,
+    uint32_t principalAccessMask,
+    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor,
+    _Out_opt_ DWORD* securityDescriptorLength
+)
+{
+    *securityDescriptor = nullptr;
+    if (securityDescriptorLength)
+    {
+        *securityDescriptorLength = 0;
+    }
+    try
```

Comment:
Consider moving to function scoped try catch (general goodness, no specific issue with code as written)